core/server/api/index.js

```diff
@@ -287,6 +295,10 @@ http = function http(apiMethod) {
             options = {};
         }
 
+        if (req.method === 'POST' && /^application\/json/.test(req.get('Content-Type')) && containsId(object)) {
```

Comment:
TBH we should probably do this for `PUT` and `PATCH` too - but that'll require changes to the admin frontend to not break it.

Reply:
I would say PUT and PATCH are fine unless the data contains an ID that doesn't match the URL `id` param.

Reply:
I think it would be worth protecting against that. Even if sending a different ID wouldn't actually change anything on a successful request I think it's good developer ergonomics - as someone using the API I'd want to know if I was sending mismatched IDs.